### PR TITLE
fix(Scripts/Midsummer): Randomize ribbon pole beam color

### PR DESCRIPTION
**Pull request by sudlud**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:

- Progress #12145

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

.event start 1
.tele orgrimmar
.go xyz 1937 -4326 22
click on ribbon pole
observe randomized beam color (blue/red/pink)
walk away from ribbon pole until beam disappears
repeat

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1695820418952974700.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- #12145 midsummer add spell script spell_midsummer_ribbon_pole_visual
+DELETE FROM acore_world.spell_script_names WHERE spell_id IN (29531, 29705, 29726, 29727);
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1695820418952974700.sql:2]: `acore_world` not needed. you already are in acore_world

File: data/sql/updates/pending_db_world/rev_1695820418952974700.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+INSERT INTO acore_world.spell_script_names (spell_id, ScriptName) VALUES
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1695820418952974700.sql:3]:
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id` IN (29531, 29705, 29726, 29727);
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
```

you don't need to specificy the table, as it is already use when adding a file to `pending_db_world` also you should use AC sql conventions

**Review by elthehablo — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1695820418952974700.sql (hunk 2 above)
Comment (sudlud) [reply data/sql/updates/pending_db_world/rev_1695820418952974700.sql:3]: thanks for the headsup, I generated them with DBeaver but should definitely follow AC convention

File: src/server/scripts/Events/midsummer.cpp [hunk 3]
```diff
@@ -218,7 +222,9 @@ class spell_midsummer_ribbon_pole : public AuraScript
             Creature* cr = target->FindNearestCreature(NPC_RIBBON_POLE_DEBUG_TARGET, 10.0f);
             if (!cr)
             {
-                target->RemoveAura(SPELL_RIBBON_POLE_CHANNEL_VISUAL);
+                target->RemoveAura(SPELL_TEST_RIBBON_POLE_CHANNEL_BLUE);
```
Comment (Kitzunu) [inline src/server/scripts/Events/midsummer.cpp:225]: You need to validate all spells used in scripts.

Comment (Kitzunu) [reply src/server/scripts/Events/midsummer.cpp:225]:
```cpp
    bool Validate(SpellInfo const* /*spell*/) override
    {
        return ValidateSpellInfo({ SPELL_CRAB_DISGUISE });
    }
```
Example taken from earlier in the file

Comment (sudlud) [reply src/server/scripts/Events/midsummer.cpp:225]: Alright will update this

**Review by elthehablo — APPROVED** [review]

**Comment by sudlud** [comment]
@Kitzunu I'll add the validatespell calls later today

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**